lib/runner.js

```diff
@@ -724,42 +729,22 @@ Runner.prototype.uncaught = function (err) {
     return;
   }
 
-  runnable.clearTimeout();
-
   // Ignore errors if already failed or pending
   // See #3226
   if (runnable.isFailed() || runnable.isPending()) {
     return;
   }
   // we cannot recover gracefully if a Runnable has already passed
   // then fails asynchronously
-  var alreadyPassed = runnable.isPassed();
-  // this will change the state to "failed" regardless of the current value
-  this.fail(runnable, err);
-  if (!alreadyPassed) {
```

Comment:
why was all of this removed?

Reply:
All this just another implementation of what is already in the tests `done()` function. So it's duplicate code.

It's trying to figure out which hook is being executed, and take certain actions depending on that. All that is already done, in a better way, in the tests `done()` function.